use travis docker env

dbu opened the pull request:
follow-up of #686 - try without installing enchant explicitly. @WouterJ is that what you ment?

dbu commented:
hm, does not work. @WouterJ can you help me?

WouterJ commented:
No. [Enchant is put on the apt packages whitelist](https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise#L519). Packages in this whitelist can be installed using the [apt addon](http://docs.travis-ci.com/user/apt/). This addon makes it possible to install packages in the container based environment.

```diff
-before_script:
-    - sudo apt-get install enchant
+sudo: false
+
+addons:
+    apt:
+        packages: [enchant]
```

dbu commented:
ah, interesting. updated the travis config file, lets see

dbu commented:
the pr build is completely broken. i guess that is the same problem lukas remarked somewhere else.

but the push also fails, with lots of spelling errors. locally, i get none of these. any idea what is happening here?

dantleech commented:
Well, none of those words are in my gcide dictionary -- either its not taking the whitelist into account or it just suddenly started detecting more words.

dbu commented:
weird. it looks like some things where previously not spell checked. there is nothing for `Multi-Language` in our wordlist.

WouterJ commented:
Builds are fixed now, if you rebase things should work.

dbu commented:
rebased. lets see...

dbu commented:
nope. it seems the enchant we end up with uses a different logic regarding spell checks in title. can that be the problem? do we need to update titles / add their words to the wordlist?

dantleech commented:
hmm, adding these words to the whitelist would solve it:

````
afterwards
balancer
balancers
boolean
filename
getters
initializer
initializers
licensor
login
lookup
multi
natively
pluggable
postfix
redirections
refactoring
runtime
subtree
subtrees
th
uncomment
workspaces
````

dbu commented:
fixed a bunch of things and added words. looks like this version of enchant has thrown out a lot of technical temrs that we needed. have a close look at the spelling list.

dbu commented on book/static_content.rst:
```diff
@@ -78,5 +78,4 @@ developers with a small and easy to understand starting point you can extend
 or use as inspiration to develop your own content types, Controllers and Admin
 panels.
 
-.. _`multilanguage support in PHPCR-ODM`: http://docs.doctrine-project.org/projects/doctrine-phpcr-odm/en/latest/reference/multilang.html
```
this was unused

dbu commented on spelling_word_list.txt:
```diff
@@ -1,14 +1,18 @@
 admin
 admins
 ajax
+afterwards
```
isn't this a pretty normal english word?

dantleech replied:
Yeah, that really should be in there (its in my gcide dictionary
locally). Anything strange about where the word occurs?

On Sat, Aug 01, 2015 at 02:09:20AM -0700, David Buchmann wrote:
>    In [1]spelling_word_list.txt:
> 
>  > @@ -1,14 +1,18 @@
>  >  admin
>  >  admins
>  >  ajax
>  > +afterwards
> 
>    isn't this a pretty normal english word?
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/symfony-cmf-docs/pull/688#discussion_r36032057
>    2. https://github.com/symfony-cmf/symfony-cmf-docs/pull/688/files#r36032057
>    3. https://github.com/symfony-cmf/symfony-cmf-docs/pull/688/files#r36032057

dbu replied:
i see nothing unusual. going to keep this one then

dbu commented on spelling_word_list.txt:
```diff
@@ -87,12 +99,15 @@ stylesheet
+subtrees
 symfony
 templating
+thats
```
is this incorrect? its used here https://github.com/symfony-cmf/symfony-cmf-docs/blob/master/tutorial/conclusion.rst

dantleech replied:
Yeah thats incorrect. It should be "that's" (it's a contraction of "that
is"). Otherwise its just plurat "that" which I am not sure ever makes
sense :P

On Sat, Aug 01, 2015 at 02:10:04AM -0700, David Buchmann wrote:
>    In [1]spelling_word_list.txt:
> 
>  >  symfony
>  >  templating
>  > +thats
> 
>    is this incorrect? its used here
>    [2]https://github.com/symfony-cmf/symfony-cmf-docs/blob/master/tutorial/conclusion.rst
> 
>    —
>    Reply to this email directly or [3]view it on GitHub.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/symfony-cmf-docs/pull/688#discussion_r36032067
>    2. https://github.com/symfony-cmf/symfony-cmf-docs/blob/master/tutorial/conclusion.rst
>    3. https://github.com/symfony-cmf/symfony-cmf-docs/pull/688/files#r36032067
>    4. https://github.com/symfony-cmf/symfony-cmf-docs/pull/688/files#r36032067

dbu commented on spelling_word_list.txt:
```diff
@@ -87,12 +102,16 @@ stylesheet
+subtrees
 symfony
 templating
+thats
```
is this incorrect? its used here https://github.com/symfony-cmf/symfony-cmf-docs/blob/master/tutorial/conclusion.rst

dbu commented:
yay, green and seems good to me. @WouterJ can you have a look and merge please?
